Gems/GradientSignal/Code/Include/GradientSignal/Editor/EditorGradientPreviewRenderer.h

```diff
@@ -249,86 +250,144 @@ namespace GradientSignal
+                    // 1 . . . 2 . . .
+                    // . . . . . . . .
+                    // . . . . . . . .
+                    // . . . . . . . . 
+                    // 3 . . . 3 . . .
+                    // . . . . . . . .
+                    // . . . . . . . .
+                    // . . . . . . . .
+                    //
+                    // Every 2 passes from then on do the same thing, with shrinking grids. One pass fills in the grid X midpoints on the
+                    // lines that were already processed, and the second pass fills in all the equivalent points on the Y grid midpoints
+                    // x . 4 . x . 4 .
+                    // . . . . . . . .
+                    // 5 . 5 . 5 . 5 .
+                    // . . . . . . . .
+                    // x . 4 . x . 4 .
+                    // . . . . . . . .
+                    // 5 . 5 . 5 . 5 .
+                    // . . . . . . . .
+                    //
+                    // x 6 x 6 x 6 x 6
+                    // 7 7 7 7 7 7 7 7
+                    // x 6 x 6 x 6 x 6
+                    // 7 7 7 7 7 7 7 7
+                    // x 6 x 6 x 6 x 6
+                    // 7 7 7 7 7 7 7 7
+                    // x 6 x 6 x 6 x 6
+                    // 7 7 7 7 7 7 7 7
+
+                    // For X, we want our starting pixel offset to alternate between 0 and a decreasing power of 2 on every pass, and
+                    // our stride to decrease by a power of 2 every two passes, ending with an offset of 0 and a stride of 1 on the last
+                    // pass.
+                    const uint64_t xOffsetShifter = AZ::GetMin(m_finalInterlacingPass - curPass, m_finalInterlacingPass - 1);
+                    const uint64_t xPixelOffset = (curPass % 2) * (1LL << (xOffsetShifter / 2));
+                    const uint64_t xPixelStride = 1LL << ((xOffsetShifter + 1) / 2);
+
+                    // For Y, we want our starting pixel offset and our stride to behave the same as X, except that we hold our starting
+                    // offset and stride for one additional pass which is what causes the first 3 passes to behave differently than the
+                    // rest. The pass offset between X and Y is also what causes the pattern to keep filling in pixels and lines that
+                    // haven't already been processed.
+                    const uint64_t yOffsetShifter = AZ::GetMin(m_finalInterlacingPass - curPass + 1, m_finalInterlacingPass - 1);
+                    const uint64_t yPixelOffset = (yOffsetShifter % 2) * (1LL << (yOffsetShifter / 2));
+                    const uint64_t yPixelStride = 1LL << ((yOffsetShifter + 1) / 2);
+
+                    // First, we loop and fill in all the gradientLookupPositions and pixelBufferIndex values for any pixels that don't
+                    // get culled out.
+                    for (uint64_t curPixel = 0; (curPixel < totalPixelsPerPass); curPixel++)
```

Comment:
nit: If this loop was rearranged a little, you could just never consider points outside the `imageBounds` instead of rejecting them. X and Y would also be simpler to calculate. Something like
```
for (uint64_t y= yPixelOffset; y < m_imageBoundsY; y += yPixelStride)
{
    for (uint64_t x= xPixelOffset; x < m_imageBoundsX; x += xPixelStride)
    {
    ....
    }
}

Reply:
lol yeah that's significantly better.  idk what i was thinking, changed!